File: techconnect/lib/models/contact_request.dart
class ContactRequest {
  final String technicianEmail;
  final String message;

  ContactRequest({
    required this.technicianEmail,
    required this.message,
  });

  Map<String, dynamic> toJson() {
    return {
      'technicianEmail': technicianEmail,
      'message': message,
    };
  }
}

File: techconnect_b/src/main/java/com/example/techconnect/backend/entity/ContactRequest.java
package com.techconnect.backend.entity;

import jakarta.persistence.*;
import java.time.LocalDateTime;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

@Entity
@Table(name = "contact_requests")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ContactRequest {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(nullable = false)
  private String userEmail;

  @Column(nullable = false)
  private String technicianEmail;

  @Column(nullable = false, length = 2000)
  private String message;

  @CreationTimestamp
  private LocalDateTime sentAt;

  private boolean read = false;
}
